### ISSUE TITLE
Type Hints: FileSource.py

### ISSUE THREAD

**Issue by austinmatherne-wk** [Arelle/Arelle#332]
Add type hints to FileSource.py

**Comment by austinmatherne-wk** [comment]
@ggravlingen I'm breaking these out into separate issues (required with how we track work internally). I think you said you were already looking at adding type hints to this one. If you are, leave a comment here so we don't accidentally duplicate work.

**Comment by ggravlingen** [comment]
Yes, I've started working on this one. I've gotten quite far but got stock on a few details. I'll push it as a draft to discuss, you can probably help me resolve the issues.

**Comment by austinmatherne-wk** [comment]
Awesome! Thank you!

**Comment by ggravlingen** [comment]
Thanks for the review help!
